src/invoke_training/training/shared/base_model_version.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+
+from transformers import PretrainedConfig
+
+
+class BaseModelVersionEnum(Enum):
+    STABLE_DIFFUSION_V1 = 1
+    STABLE_DIFFUSION_V2 = 2
+    STABLE_DIFFUSION_SDXL_BASE = 3
+    STABLE_DIFFUSION_SDXL_REFINER = 4
+
+
+def get_base_model_version(
+    diffusers_model_name: str, revision: str = "main", local_files_only: bool = True
+) -> BaseModelVersionEnum:
+    """Returns the `BaseModelVersionEnum` of a diffusers model.
+
+    Args:
+        diffusers_model_name (str): The diffusers model name (on Hugging Face Hub).
+        revision (str, optional): The model revision (branch or commit hash). Defaults to "main".
+
+    Raises:
+        Exception: If the base model version can not be determined.
+
+    Returns:
+        BaseModelVersionEnum: The detected base model version.
+    """
+    unet_config = PretrainedConfig.from_pretrained(
+        pretrained_model_name_or_path=diffusers_model_name,
+        revision=revision,
+        subfolder="unet",
+        local_files_only=local_files_only,
+    )
+
+    # This logic was copied from
+    # https://github.com/invoke-ai/InvokeAI/blob/e77400ab62d24acbdf2f48a7427705e7b8b97e4a/invokeai/backend/model_management/model_probe.py#L412-L421
+    # This seems fragile. If you see this and know of a better way to detect the base model version, your contribution
```

Comment:
This seems to be the standard way of doing it right now unfortunately. To specifically point out xl vs not xl, the diffusers class name is pretty clear in the root config of the model. Beyond that, I'm not sure of a better way.